tasks/dataset-split/main.py

```diff
@@ -0,0 +1,157 @@
+from pathlib import Path
+
+import sys
+import csv
+import zipfile
+import logging
+
+from coretex import currentTaskRun
+from coretex import NetworkDataset, ImageDataset, CustomDataset, SequenceDataset
+
+
+def numberOfSamplesInEachNewDataset(n: int, numberOfNewDatasets: int) -> list[int]:
+    r = n % numberOfNewDatasets
+    numberOfSamples = [n // numberOfNewDatasets] * numberOfNewDatasets
+    for i in range(r):
+        numberOfSamples[i] += 1
+
+    return numberOfSamples
+
+
+def imageDatasetSplit(dataset: ImageDataset, numberOfSamples: list[int], projectID: int) -> list[NetworkDataset]:  
+    dataset.download()
+    samples = dataset.samples
+    listOfNewDatasets: list[NetworkDataset] = []
+
+    numberOfNewDatasets = len(numberOfSamples)
+    
+    counter = 0
+    for i in range(numberOfNewDatasets):
+        newDataset = ImageDataset.createDataset(f"{currentTaskRun().id}-newDataset-{i}", projectID)
```

Comment:
currentTaskRun should only be called once, at the beginning of the task. The returned TaskRun object can then be passed to other functions that use it.